src/main/java/net/glowstone/block/itemtype/ItemType.java

```diff
@@ -130,15 +130,17 @@ protected final void setMaxStackSize(int maxStackSize) {
      * @param holding The ItemStack the player was holding
      */
     public void rightClickAir(GlowPlayer player, ItemStack holding) {
-        // nothing by default
+        if (fastEquip) {
```

Comment:
This will cause dupe usage, as the client sends a right click and use packet. This is not meant to check if self usage is only allowed.